packages/eas-cli/src/credentials/manager/ManageAndroid.ts

```diff
@@ -140,29 +147,26 @@ export class ManageAndroid {
     }
   }
 
-  private async createProjectContextAsync(ctx: CredentialsContext): Promise<{
+  private async createProjectContextAsync(
+    ctx: CredentialsContext,
+    { buildProfile }: { buildProfile?: BuildProfile<Platform.ANDROID> }
+  ): Promise<{
     gradleContext?: GradleBuildContext;
-    buildProfile?: BuildProfile<Platform.ANDROID>;
   }> {
     if (!ctx.hasProjectContext) {
       return {};
     }
 
+    assert(buildProfile, 'buildProfile must be defined in project context');
```

Comment:
Hmm, I know it wasn't added in this PR but I think the `if` + the new assert looks weird. I'd move the `if` out of the function and only run `createProjectContextAsync` if we have the project context.

Reply:
Sounds good. I've also changed similar logic in `ManageIos`